### PR TITLE
Migrate plugin packaging tests to java

### PR DESCRIPTION
**Pull request by rjernst**
This commit converts the bats tests for the plugin cli into the java
packaging test framework. The new tests only use the example plugin to
test the plugin cli. The tests for each individual plugin's contents
after being installed are handled by a new unit test for the plugin
installer added in #58287.

### CODE REVIEW

**Review by mark-vieira — APPROVED** [review]
The build bits all seem sound to me but someone more familiar might want to take a pass at the actual test coverage.

File: qa/os/src/test/java/org/elasticsearch/packaging/util/Shell.java [hunk 1]
```diff
@@ -200,9 +216,9 @@ private Result runScriptIgnoreExitCode(String[] command) {
     private String readFileIfExists(Path path) throws IOException {
         if (Files.exists(path)) {
             long size = Files.size(path);
-            if (size > 100 * 1024) {
+            /*if (size > 100 * 1024) {
```
Comment (nik9000) [inline qa/os/src/test/java/org/elasticsearch/packaging/util/Shell.java:219]: Leftover?

File: qa/os/src/test/java/org/elasticsearch/packaging/test/PluginCliTests.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                assertThat(settingsResponse, equalTo("{\"defaults\":{\"custom\":{\"simple\":\"foo\"}}}"));
+            });
+        });
+
+        Files.delete(pluginsDir);
+        Files.move(stashedPluginsDir, pluginsDir);
+    }
+
+    public void test21CustomConfDir() throws Exception {
+        withCustomConfig(confPath -> assertWithExamplePlugin(installResult -> {}));
```
Comment (nik9000) [inline qa/os/src/test/java/org/elasticsearch/packaging/test/PluginCliTests.java:97]: I don't remember why some of them booted ES and asserted that the example plugin looks good and some didn't. I think what you have here is fine though, just asserting that the installer didn't blow up is probably good enough in these cases so long as we have a single test that asserts that it did indeed install the plugin in all the way. And you have that above.

:+1:

**Review by nik9000 — CHANGES_REQUESTED** [review]
LGTM except I think you commented out some code you meant to keep.

File: qa/os/src/test/java/org/elasticsearch/packaging/util/Shell.java (hunk 1 above)
Comment (rjernst) [reply qa/os/src/test/java/org/elasticsearch/packaging/util/Shell.java:219]: Oops yes, will uncomment.

Comment (rjernst) [reply qa/os/src/test/java/org/elasticsearch/packaging/util/Shell.java]: I pushed d6b6627

**Review by nik9000 — APPROVED** [review]
LGTM

**Pull request merged**